consensus/src/dag/tests/order_rule_tests.rs

```diff
@@ -0,0 +1,142 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    dag::{
+        anchor_election::RoundRobinAnchorElection,
+        dag_store::Dag,
+        order_rule::OrderRule,
+        tests::{dag_test::MockStorage, helpers::new_certified_node},
+        types::NodeCertificate,
+        CertifiedNode,
+    },
+    test_utils::placeholder_ledger_info,
+};
+use aptos_consensus_types::common::Author;
+use aptos_infallible::RwLock;
+use aptos_types::{
+    aggregate_signature::AggregateSignature, epoch_state::EpochState,
+    validator_verifier::random_validator_verifier,
+};
+use futures_channel::mpsc::{unbounded, UnboundedReceiver};
+use proptest::prelude::*;
+use std::sync::Arc;
+
+/// Generate a virtual dag that first layer represents round
+/// second layer represents nodes
```

Comment:
need to fix this comment? I only see two layers.

Reply:
it's three nested vector :)